### DISCUSSION TITLE
thermal conductivity and diffusivity specification for fluid and solid

### DISCUSSION THREAD

**Discussion by sanjeev784** [xcompact3d/Incompact3d#146] [Q&A] [answered]
Hi,

Can anybody help me to specify different thermal conductivity and diffusivity to both solid and fluid using IBM=1. Thanks in advance for the help.

Regards,
Sanjeev

**Comment by mathrack** [comment, marked as answer]
Hello,

I think the IBM (in the current implementation) does not allow fluid-solid thermal coupling.

Best regards,
Cédric

**Reply by sanjeev784** [reply]
Hi Cédric

Thanks for the reply.

So Is there any way to solve conjugate heat transfer problem in the current implementation.

Regards,
Sanjeev

**Reply by rvicentecruz** [reply]
Hello, 

We are currently working on that, a version of the solver including fluid-solid thermal coupling in pipe flow should be available soon (however for IBM=2). If you want more details on the method, I can refer you to [chapter 6 of my thesis manuscript](https://www.theses.fr/2021POIT2302) and [to the following article](http://dx.doi.org/10.1016/j.compfluid.2021.105101) (plane channel configuration in Incompact3d).

Best regards,
Rodrigo
